lua/tests/infoview/run_spec.lua

```diff
@@ -83,128 +71,102 @@ describe('infoview', function()
-      assert.is.equal(num_wins - 1, get_num_wins())
-      assert.is.equal(win, vim.api.nvim_get_current_win())
-      assert.is_not.open_infoview()
+      assert.closed_infoview()
     end)
   end)
 
-  infoview_info = infoview.get_current_infoview():open()
-
-  vim.api.nvim_command("tabedit lua/tests/fixtures/example-lean4-project/Test.lean")
-
-  local new_buf = vim.api.nvim_get_current_buf()
-  local new_win = vim.api.nvim_get_current_win()
-  local new_infoview_info = infoview.get_current_infoview():open()
-
   describe('new tab', function()
     it('closes independently',
     function(_)
-      local num_wins = get_num_wins()
+      infoview.get_current_infoview():open()
+      assert.opened_infoview()
+      vim.api.nvim_command("tabnew")
+      assert.created_win()
+      vim.api.nvim_command("edit lua/tests/fixtures/example-lean4-project/Test.lean")
+      assert.opened_infoview()
       infoview.get_current_infoview():close()
-      assert.is_true(vim.api.nvim_win_is_valid(infoview_info.window))
-      assert.is_false(vim.api.nvim_win_is_valid(new_infoview_info.window))
-      assert.is.equal(num_wins - 1, get_num_wins())
-      assert.is_not.open_infoview()
+      assert.closed_infoview()
+      assert.is_not.update_enabled()
+      vim.api.nvim_command("tabprevious")
+      assert.opened_infoview_kept()
     end)
 
     it('CursorHold(I) disabled independently',
     function(_)
-      assert.is_not.update_enabled()
-      vim.api.nvim_set_current_win(win)
       assert.is.update_enabled()
     end)
 
     it('CursorHold(I) updated on BufEnter',
     function(_)
-      vim.api.nvim_set_current_buf(new_buf)
+      vim.api.nvim_command("edit lua/tests/fixtures/example-lean4-project/Test.lean")
+      assert.opened_infoview_kept()
       assert.is.update_enabled()
     end)
 
     it('CursorHold(I) updated on WinEnter',
     function(_)
-      vim.api.nvim_set_current_win(new_win)
+      vim.api.nvim_command("tabnext")
+      assert.closed_infoview_kept()
       assert.is_not.update_enabled()
     end)
 
     it('remains closed on BufWinEnter',
     function(_)
-      local num_wins = get_num_wins()
       vim.api.nvim_command("edit lua/tests/fixtures/example-lean4-project/Test/Test1.lean")
-      assert.is.equal(num_wins, get_num_wins())
+      assert.closed_infoview_kept()
     end)
 
     it('opens independently',
     function(_)
-      vim.api.nvim_set_current_win(win)
+      vim.api.nvim_command("tabprevious")
       infoview.get_current_infoview():close()
-      vim.api.nvim_set_current_win(new_win)
-      local num_wins = get_num_wins()
-      new_infoview_info = infoview.get_current_infoview():open()
-      assert.is_false(vim.api.nvim_win_is_valid(infoview_info.window))
-      assert.is_true(vim.api.nvim_win_is_valid(new_infoview_info.window))
-      assert.is.equal(num_wins + 1, get_num_wins())
-      assert.open_infoview()
+      assert.closed_infoview()
+      vim.api.nvim_command("tabnext")
```

Comment:
TODO: would be good form to do `assert.changed_win()` here (and below), but it's actually not an issue in this specific case because we aren't actually closing anything.